samples/nvidia-resnet/pipeline/pipeline.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import kfp.dsl as dsl
+import datetime
+import os
+from kubernetes import client as k8s_client
+
+
+PIPELINE_NAME = 'resnet_cifar10_pipeline'
+
+
+def preprocess_op(persistent_volume_path, input_dir, output_dir, step_name='preprocess'):
+    return dsl.ContainerOp(
+        name=step_name,
+        image='nvcr.io/nvidian/sae/ananths:kubeflow-preprocess',
+        command=['python'],
```

Comment:
Writing something along these lines will add the `processed_data_dir` output to this component which can then be passed to the next component.

```python
command=[
  'sh', '-c', 'echo "$0" > "$1"; "$*"', output_dir, '/tmp/output_dir',  
  'python',
],
file_outputs={'processed_data_dir': '/tmp/output_dir'}
```

Reply:
Updated.